doc/sailfish-webview-webviewadditional.qdoc

```diff
@@ -0,0 +1,321 @@
+controller, a default TextSelectionController from Sailfish.WebView.Controls
+will be constructed and used when required.
+
+Any text selection controller provided by the client must conform
+to the required API.  See the documentation for\c{Sailfish.WebView.Controls}
+for more information.
+
+\section2 WebView::downloadsEnabled
+
+\c{bool}-type property.
+
+Whether downloads are enabled within the webview.
+
+If downloads are enabled, long-press context menus will include options to
+save the link or image to the device, when appropriate.
+
+\section2 WebView::virtualKeyboardMargin
+
+\c{qreal}-type property.
+
+The margin taken by the keyboard at the bottom of the page.
+
+When using a \l{WebView} this property will be set automatically when the
+keyboard is opened or closed, and as such, most developers will never need
+to change the default behavior by setting this property manually.
+
+In some special cases, if the application developer wishes to control
+precisely how web content should be arranged to compensate for the open
+keyboard (e.g. by allowing the keyboard to overlap the web content to some
+extent, for some reason), they can set this value manually.
+
+\section2 WebView::orientation
+
+\c{Qt::ScreenOrientation}-type property.
+
+The orientation of the webview.
+
+Ordinarily, the value of this property will follow the orientation of the
+(Silica) Page in which the web view is shown.
+
+\section2 WebView::viewportWidth
+
+\c{qreal}-type property.
+
+The width of the webview viewport, in device pixels.
+
+Note that the viewport width will usually be the same as the width of the
+web view, but in some cases clients may wish to set the viewport width
+explicitly (e.g. to show some chrome at one edge or the other, without
+obscuring some content underneath, while keeping the geometry of the web view
+item itself unchanged).
+
+Device orientation should be considered when binding to this property.
+
+See also: \l{WebView::contentRect}, \l{WebView::orientation}
+
+\section2 WebView::viewportHeight
+
+\c{qreal}-type property.
+
+The height of the webview viewport, in device pixels, which by default is
+bound to the screen height (in portrait orientation) or width (in landscape
+orientation).
+
+Note that the viewport height will usually be the same as the height of the
+web view, but in some cases clients may wish to set the viewport height
+explicitly (e.g. to show some chrome at one edge or the other, without
+obscuring some content underneath, while keeping the geometry of the web view
+item itself unchanged).
+
+Device orientation should be considered when binding to this property.
+
+See also: \l{WebView::contentRect}, \l{WebView::orientation}
+
+\section2 WebView::atXBeginning
+
+\c{bool}-type read-only property.
+
+Whether the viewport of the webview is scrolled to the far-left of the content.
+
+Effectively, the value of this property will be true if the
+\l{WebView::scrollableOffset} x-value is approximately zero.
```

Comment:
Why "approximately" (I just wonder if it's worth explaining)?

Reply:
done